src/config.rs

```diff
@@ -65,10 +67,53 @@ impl Deserialize for store::Store {
 }
 
 
-/// Implementation with single method to read configuration from JSON.
+impl Deserialize for Template {
+    fn deserialize<D>(de: &mut D) -> Result<Template, D::Error>
+                      where D: serde::Deserializer {
+        let value: Value = try!(serde::Deserialize::deserialize(de));
+        let path = try!(value.as_str().ok_or_else(|| {
+            serde::de::Error::custom("template path is not a string")
+        }));
+        match mustache::compile_path(path) {
+            Ok(tmpl) => Ok(Template(tmpl)),
+            Err(err) => Err(serde::de::Error::custom(err.description())),
+        }
+    }
+}
+
+
+impl Template {
+    pub fn render(&self, params: &[(&str, &str)]) -> String {
+        let mut builder = mustache::MapBuilder::new();
+        for &param in params {
+            let (ref key, ref value) = param;
+            builder = builder.insert_str(key, value);
+        }
+        let mut out: Vec<u8> = Vec::new();
+        self.0.render_data(&mut out, &builder.build());
+        String::from_utf8(out).unwrap()
+    }
+}
+
+
+
+impl Templates {
+    fn default_forward() -> Template {
+        Template(mustache::compile_path("tmpl/forward.mustache").unwrap())
```

Comment:
It makes more sense for me to inline this into the `default_templates()` method, below.